### ISSUE TITLE
Cannot remove users 'entirely' from cognito user pool with admin_delete_user

### ISSUE THREAD

**Issue by arpekkar** [localstack/localstack#3754]
# Type of request: This is a ...

[x] bug report
[ ] feature request

# Detailed description
Since version 0.12.6 I have not been able to remove users from cognito user pool in
my integration tests. I get this error: "An error occurred (400) when
calling the AdminCreateUser operation: An account with the given
username already exists".

My integration tests are initialized so that I first create user pool for
the session and then for each function I create new user to the user
pool and after test I delete that user from the pool in following way:

## Expected behavior

I should be able to create users with **admin_create_user**
users should be able to be removed entirely from the user pool with **admin_delete_user**
without destroying the whole user pool first

## Actual behavior
User is not removed entirely with **admin_delete_user**
If I then run my test again by trying to create new user I get:

"An error occurred (400) when
calling the AdminCreateUser operation: An account with the given
username already exists".

# Steps to reproduce

## Client code (AWS SDK code snippet, or sequence of "awslocal" commands)

First create user pool in conftest.py with scope="session":

```
integration_test_user_pool = integration_test_cognito.create_user_pool(
    PoolName=user_pool_name  # 'integration-test-user-pool'
)
```

Then in scope="function" create user:

```
cognito_user_1 = integration_test_cognito.admin_create_user(
        UserPoolId=integration_test_user_pool["UserPool"]["Id"],
        Username=integration_test_dataset.user_1.email,
        UserAttributes=[
            {"Name": "email", "Value": integration_test_dataset.user_1.email}
        ],
   )

```

and finally  delete:

```
cognito_users = integration_test_cognito.list_users(
        UserPoolId=integration_test_user_pool["UserPool"]["Id"]
    )
for user in cognito_users["Users"]:
        integration_test_cognito.admin_delete_user(
            UserPoolId=integration_test_user_pool["UserPool"]["Id"],
            Username=user["Username"],
        )
```

It seems to delete them in the tear down, but once I start creating the same user with same email for next test, it gives the error above "An error occurred (400) when
calling the AdminCreateUser operation: An account with the given
username already exists" - as if there still would be a record of that user in the pool

Basically I could workaround the issue just by deleting the whole user pool and creating a new one for each test, but just wondering what is happening here. Am I doing something wrong or is there some issues in the localstack?

**Comment by whummer** [comment]
Thanks for reporting @arpekkar . This should (hopefully) be fixed in the meantime - can you please pull the `latest` Docker image and give it another try? Please report here if the problem persists. Thanks for your help!
